src/utils/components/DiskModal/components/DiskSourceSelect/components/DiskSourceDataSourceSelect/DiskSourceDataSourceSelectNamespace.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { FC } from 'react';
+import { Controller, FieldPath, useFormContext } from 'react-hook-form';
+
+import { modelToGroupVersionKind, ProjectModel } from '@kubevirt-ui/kubevirt-api/console';
+import { V1DiskFormState } from '@kubevirt-utils/components/DiskModal/utils/types';
+import InlineFilterSelect from '@kubevirt-utils/components/FilterSelect/InlineFilterSelect';
+import FormGroupHelperText from '@kubevirt-utils/components/FormGroupHelperText/FormGroupHelperText';
+import Loading from '@kubevirt-utils/components/Loading/Loading';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import useProjects from '@kubevirt-utils/hooks/useProjects';
+import { FormGroup, ValidatedOptions } from '@patternfly/react-core';
+
+import {
+  DATAVOLUME_DATASOURCE_NAME,
+  DATAVOLUME_DATASOURCE_NAMESPACE,
+} from '../../../utils/constants';
+import { getDataSourceNamespaceError } from '../../../utils/selectors';
+import { diskSourceDataSourceNamespaceFieldID } from '../../utils/constants';
+
+const DiskSourceDataSourceSelectNamespace: FC = () => {
+  const { t } = useKubevirtTranslation();
+  const {
+    control,
+    formState: { errors },
+    setValue,
+  } = useFormContext<V1DiskFormState>();
+  const [projectNames, projectsLoaded] = useProjects();
+
+  if (!projectsLoaded) return <Loading />;
+
+  const error = getDataSourceNamespaceError(errors);
+
+  return (
+    <Controller
+      render={({ field: { onChange, value } }) => (
+        <FormGroup
+          fieldId={diskSourceDataSourceNamespaceFieldID}
+          id={diskSourceDataSourceNamespaceFieldID}
+          isRequired
+          label={t('DataSource project')}
+        >
+          <InlineFilterSelect
+            options={projectNames?.map((name) => ({
```

Comment:
i would put it in a useMemo , as other things from from hook can cause it to rerun